### PR TITLE
[qa] Fix rare failure in p2p-segwit.py

### LINKED ISSUE #11953: travis failure in p2p-segwit.py not(b'bad-txns-vout-negative' == b'bad-witness-nonstandard')
This keeps popping up from time to time on travis. The relevant log excerpt:

```
2017-12-19 17:52:57.940000 TestFramework (INFO): Testing detection of non-standard P2WSH witness
2017-12-19 17:52:58.710000 TestFramework (ERROR): Assertion failed
Traceback (most recent call last):
  File "./test/functional/test_framework/test_framework.py", line 118, in main
    self.run_test()
  File "./test/functional/p2p-segwit.py", line 1943, in run_test
    self.test_non_standard_witness()
  File "./test/functional/p2p-segwit.py", line 1827, in test_non_standard_witness
    test_transaction_acceptance(self.nodes[1].rpc, self.std_node, p2wsh_txs[0], True, False, b'bad-witness-nonstandard')
  File "./test/functional/p2p-segwit.py", line 48, in test_transaction_acceptance
    assert_equal(p2p.last_message["reject"].reason, reason)
  File "./test/functional/test_framework/util.py", line 38, in assert_equal
    raise AssertionError("not(%s)" % " == ".join(str(arg) for arg in (thing1, thing2) + args))
AssertionError: not(b'bad-txns-vout-negative' == b'bad-witness-nonstandard')
```

Most recent example: https://travis-ci.org/bitcoin/bitcoin/jobs/318711688

### PR DESCRIPTION
**Pull request by sdaftuar**
Avoid creating very small utxos that would violate an assumption in
test_non_standard_witness.

Fixes #11953

### CODE REVIEW

File: test/functional/p2p-segwit.py [hunk 1]
```diff
@@ -1396,18 +1396,20 @@ def test_signature_version_1(self):
...
+        tx.vout = [CTxOut(output_value, CScript([OP_TRUE]))] * 2
         for i in temp_utxos:
             # Just spend to our usual anyone-can-spend output
             # Use SIGHASH_SINGLE|SIGHASH_ANYONECANPAY so we can build up
             # the signatures as we go.
             tx.vin.append(CTxIn(COutPoint(i.sha256, i.n), b""))
-            tx.vout.append(CTxOut(i.nValue, CScript([OP_TRUE])))
             tx.wit.vtxinwit.append(CTxInWitness())
-            sign_P2PK_witness_input(witness_program, tx, index, SIGHASH_SINGLE|SIGHASH_ANYONECANPAY, i.nValue, key)
+            sign_P2PK_witness_input(witness_program, tx, index, SIGHASH_ALL|SIGHASH_ANYONECANPAY, i.nValue, key)
```
Comment (instagibbs) [inline test/functional/p2p-segwit.py]: why this change?

Comment (sdaftuar) [reply test/functional/p2p-segwit.py]: It felt wrong to use SIGHASH_SINGLE with an out-of-bounds index.

Comment (instagibbs) [reply test/functional/p2p-segwit.py]: ah, ok then comment could use an update a few lines up.

Comment (sdaftuar) [reply test/functional/p2p-segwit.py]: Oops!  Thanks, will fix.

**Comment by jnewbery** [comment]
Tested ACk 35c2b1fe3b1a046bbd1bfaf1c178e838d047277a

**Comment by MarcoFalke** [comment]
Thanks!

utACK 35c2b1fe3b1a046bbd1bfaf1c178e838d047277a. Created a similar fix last year, but somehow forgot to submit it.

**Review by ryanofsky — COMMENTED** [review]
utACK 35c2b1fe3b1a046bbd1bfaf1c178e838d047277a. Change makes sense, but I couldn't figure out what problem the previous outputs caused in test_non_standard_witness.

**Comment by MarcoFalke** [comment]
In `test_non_standard_witness`, you subtract the fee at some point. Having really small outputs, this could lead to negative outputs, provoking the "bad-txns-vout-negative". The smallest observed value depends on the test run, so this is hit very rarely.

**Comment by instagibbs** [comment]
Yes the PR comment makes it sound like it's a non-standard witness, rather than non-standard output.

**Comment by laanwj** [comment]
utACK 35c2b1f
